addons/web/static/src/webclient/actions/action_service.js

```diff
@@ -351,8 +351,10 @@ export function makeActionManager(env, router = _router) {
             }
             return Object.assign({}, action);
         }
-
         // actionRequest is an object describing the action
+        if (actionRequest.help) {
+            actionRequest.help = markup(actionRequest.help);
+        }
```

Comment:
this is wrong. You can't markup without knowing where the action comes from and whether it is safe. We markup when we read the action in db (from its id, xmlid) because it is an html field, so we know it is safe (it is sanitized server-side). If you execute an action given as an object, it's your responsibility to markup the help, if you know it is safe

Reply:
Is it possible to sanitise the object before attempting to parse it? If not, where can I move the markup call to make sure it safely covers action objects that aren't triggered by buttons?

Reply:
Where do you call doAction? I don't understand what needs to be parsed

Reply:
This is my first incursion into the web module, so I'm not really sure what's the best way to manage this, but the problem is that [the help string](https://github.com/odoo/odoo/blob/master/addons/stock/models/stock_quant.py#L1284-L1288) for inventory forecast reports is rendered as raw, unparsed HTML because the flow for the action object never reaches `markup()` as it enters from `doAction()`, unlike the way XML ids and numbers are handled (in `_loadAction()`) or the way action objects are handled when `doActionButton()` is the entry point.

Reply:
Here's where you should do the markup: https://github.com/odoo/odoo/blob/master/addons/stock/static/src/stock_forecasted/forecasted_header.js#L20

There, you know which method you call, and you know you can trust the result.

(assuming I correctly spotted the place where the doAction is done with the informations you provided me)

Reply:
Thank you! That's exactly what I was looking for.